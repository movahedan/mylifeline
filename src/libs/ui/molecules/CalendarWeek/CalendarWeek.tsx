import { twMerge } from "tailwind-merge";

import { classNames } from "@mylifeline/utilities";

import type { CSSProperties } from "react";

export type CalendarWeekProps = {
  dataTestId?: string;
  isPassed?: boolean;
  hasEvent?: boolean;
  style?: CSSProperties;
  className?: string;
};

export const CalendarWeek = ({
  dataTestId = "week",
  isPassed = false,
  hasEvent = false,
  style,
  className,
}: CalendarWeekProps) => {
  return (
    <li
      data-testid={dataTestId}
      style={style}
      className={classNames([
        "w-8 h-8 mx-1 my-1",
        twMerge(
          "bg-slate-400",
          isPassed && "bg-green-500",
          hasEvent && "bg-purple-500",
          className
        ),
      ])}
    />
  );
};
