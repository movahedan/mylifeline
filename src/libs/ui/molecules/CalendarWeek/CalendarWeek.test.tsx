import { render } from "@testing-library/react";

import { CalendarWeek } from "./CalendarWeek";

import type { CalendarWeekProps } from "./CalendarWeek";

describe("CalendarWeek", () => {
  let props: CalendarWeekProps = {};
  const dataTestId = "week";

  beforeEach(() => {
    props = {};
  });

  test("should be rendered correctly", () => {
    const { container, queryByTestId } = render(<CalendarWeek {...props} />);
    expect(container).toMatchSnapshot("default");
    expect(queryByTestId(dataTestId)).toHaveClass("bg-slate-400");
    expect(queryByTestId(dataTestId)).not.toHaveClass("bg-green-500");
    expect(queryByTestId(dataTestId)).not.toHaveClass("bg-purple-500");
  });

  test("passed week should be green", () => {
    props.isPassed = true;
    const { container, queryByTestId } = render(<CalendarWeek {...props} />);
    expect(container).toMatchSnapshot("passed week");
    expect(queryByTestId(dataTestId)).toHaveClass("bg-green-500");
  });

  test("week which has event should be purple, doesn't matter it is passed or not", () => {
    props.hasEvent = true;
    const { container, rerender, queryByTestId } = render(
      <CalendarWeek {...props} />
    );
    expect(container).toMatchSnapshot("has event week");
    expect(queryByTestId(dataTestId)).toHaveClass("bg-purple-500");
    expect(queryByTestId(dataTestId)).not.toHaveClass("bg-green-500");

    props.isPassed = true;
    rerender(<CalendarWeek {...props} />);
    expect(container).toMatchSnapshot("passed week which has event");
    expect(queryByTestId(dataTestId)).toHaveClass("bg-purple-500");
    expect(queryByTestId(dataTestId)).not.toHaveClass("bg-green-500");
  });
});
