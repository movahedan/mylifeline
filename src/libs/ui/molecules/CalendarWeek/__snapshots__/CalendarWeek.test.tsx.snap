// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarWeek passed week should be green: passed week 1`] = `
<div>
  <li
    class="w-8 h-8 mx-1 my-1 bg-green-500"
    data-testid="week"
  />
</div>
`;

exports[`CalendarWeek should be rendered correctly: default 1`] = `
<div>
  <li
    class="w-8 h-8 mx-1 my-1 bg-slate-400"
    data-testid="week"
  />
</div>
`;

exports[`CalendarWeek week which has event should be purple, doesn't matter it is passed or not: has event week 1`] = `
<div>
  <li
    class="w-8 h-8 mx-1 my-1 bg-purple-500"
    data-testid="week"
  />
</div>
`;

exports[`CalendarWeek week which has event should be purple, doesn't matter it is passed or not: passed week which has event 1`] = `
<div>
  <li
    class="w-8 h-8 mx-1 my-1 bg-purple-500"
    data-testid="week"
  />
</div>
`;
